src/components/PopupWithForm.js

```diff
@@ -24,24 +25,23 @@ export class PopupWithForm extends Popup {
   }
 
   // _getInputValue should be called to get the object.
+  setButtonText(text) {
+    // esta funcion cambia el texto del botton para submmit, puedo poner el texto que quiera como parametro
+    this._submitButton.textContent = text;
+  }
 
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (event) => {
       event.preventDefault();
+
       this._handleFormSubmit(this._getInputValues());
-      super.close();
     });
   }
 
   close() {
     this._popupForm.reset();
     super.close(); // super se refiere al padre, osea el method close del padre
+    this._submitButton.textContent = this._originalButtonText; // lo vuelve al texto original cuando cierra el popUp
```

Comment:
Needs correction: please don't change the button text here, you should do it in index.js after the api request in `finally` block